### PR TITLE
fresher GHA workflows

### LINKED ISSUE #72: Update GHA workflows
- to use the r-lib/actions v2;
- to have `GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}`.

I'll make a PR.

### PR DESCRIPTION
**Pull request by maelle**
Fix #72

### CODE REVIEW

File: .github/workflows/R-CMD-check.yaml [hunk 1]
```diff
@@ -13,90 +23,40 @@ jobs:
...
 
     env:
-      R_REMOTES_NO_ERRORS_FROM_WARNINGS: true
-      RSPM: ${{ matrix.config.rspm }}
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+      R_KEEP_PKG_SOURCE: yes
       _R_CHECK_TESTS_NLINES_: 0
 
     steps:
-      - name: Configure git
```
Comment (maelle) [inline .github/workflows/R-CMD-check.yaml:31]: Why was this step needed? Should I add it back?

File: .github/workflows/R-CMD-check.yaml [hunk 2]
```diff
@@ -13,90 +23,40 @@ jobs:
       matrix:
         config:
           - {os: macOS-latest,   r: 'release'}
+
           - {os: windows-latest, r: 'release'}
+          # Use 3.6 to trigger usage of RTools35
```
Comment (nealrichardson) [inline .github/workflows/R-CMD-check.yaml:28]: Do you think we need to mess with rtools35 here? There's no compiled code.

Comment (maelle) [reply .github/workflows/R-CMD-check.yaml]: I've now removed it.

File: .github/workflows/R-CMD-check.yaml (hunk 1 above)
Comment (maelle) [reply .github/workflows/R-CMD-check.yaml:31]: Yes all green but I can add it back with a comment if you prefer.

Comment (nealrichardson) [reply .github/workflows/R-CMD-check.yaml:31]: Nah, we should delete if it's no longer needed.

**Review by nealrichardson — APPROVED** [review]
Thanks so much for doing this!

**Pull request merged**